File: 118.ts
function generate(numRows: number): number[][] {
    if(numRows < 1)
    {
        return [[]];
    }
    else if(numRows === 1)
    {
        return [[1]];
    }
    else if(numRows === 2)
    {
        return [[1], [1,1]];
    }

    let rows: Array<Array<number>> = [[1],[1,1]];
    for(let i = 3; i <= numRows; i++)
    {
        let newRow = [1];
        for(let j = 1; j <= i-2; j++)
        {
            newRow.push(rows[i-2][j-1] + rows[i-2][j]);
        }
        newRow.push(1);
        rows.push(newRow);
    }

    return rows;
};

let testCases: Array<Array<any>> = [ [5, [[1],[1,1],[1,2,1],[1,3,3,1],[1,4,6,4,1]]], [1, [[1]]] ];
for(const testcase of testCases)
{
    console.log(`Input: ${testcase[0]}
        Expected output: ${JSON.stringify(testcase[1])}
        Actual output: ${JSON.stringify(generate(testcase[0]))}
        `);
}

File: 118.py
from typing import List

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows < 1:
            return []
        elif numRows == 1:
            return [[1]]
        elif numRows == 2:
            return [ [1], [1, 1] ]

        rows = [ [1], [1, 1] ]
        for i in range(3, numRows + 1):
            row = [1]
            for j in range(2, i):
                row.append(rows[i-2][j-2] + rows[i-2][j-1])
            row.append(1)
            rows.append(row)

        return rows

if __name__ == '__main__':
    test_cases = ( (5, [[1],[1,1],[1,2,1],[1,3,3,1],[1,4,6,4,1]]), (1, [[1]]))
    for case in test_cases:
        print('Input: ' + str(case[0]) + '\nExpected output: ' + str(case[1]) + '\nActual output: ' + str(Solution().generate(case[0])) + '\n')